authorization/resource/resource_type_scope_blackbox_test.go

```diff
@@ -0,0 +1,154 @@
+package resource_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-auth/authorization/resource"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/gormsupport/cleaner"
+	"github.com/fabric8-services/fabric8-auth/gormtestsupport"
+	"github.com/fabric8-services/fabric8-auth/migration"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/satori/go.uuid"
+
+	res "github.com/fabric8-services/fabric8-auth/resource"
+)
+
+type resourceTypeScopeBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	repo             resource.ResourceTypeScopeRepository
+	resourceTypeRepo resource.ResourceTypeRepository
+	clean            func()
+	ctx              context.Context
+}
+
+func TestRunResourceTypeScopeBlackBoxTest(t *testing.T) {
+	suite.Run(t, &resourceTypeScopeBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite("../../config.yaml")})
+}
+
+// SetupSuite overrides the DBTestSuite's function but calls it before doing anything else
+// The SetupSuite method will run before the tests in the suite are run.
+// It sets up a database connection for all the tests in this suite without polluting global space.
+func (s *resourceTypeScopeBlackBoxTest) SetupSuite() {
+	s.DBTestSuite.SetupSuite()
+	s.ctx = migration.NewMigrationContext(context.Background())
+	s.DBTestSuite.PopulateDBTestSuite(s.ctx)
+}
+
+func (s *resourceTypeScopeBlackBoxTest) SetupTest() {
+	s.DB.LogMode(true)
+	s.repo = resource.NewResourceTypeScopeRepository(s.DB)
+	s.resourceTypeRepo = resource.NewResourceTypeRepository(s.DB)
+	s.clean = cleaner.DeleteCreatedEntities(s.DB)
+}
+
+func (s *resourceTypeScopeBlackBoxTest) TearDownTest() {
+	s.clean()
+}
+
+func (s *resourceTypeScopeBlackBoxTest) TestOKToDelete() {
+	t := s.T()
+	res.Require(t, res.Database)
```

Comment:
You don't have to require Database explicitly in gormtestsupport.DBTestSuite test suites. It's already required.